packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/RelationModal.tsx

```diff
@@ -71,16 +84,38 @@ const RelationModalWrapper = ({ relation, triggerButtonLabel }: RelationModalPro
   );
   const currentDocumentMeta = useDocumentContext('RelationModalBody', (state) => state.meta);
   const changeDocument = useDocumentContext('RelationModalBody', (state) => state.changeDocument);
+  const documentHistory = useDocumentContext('RelationModalBody', (state) => state.documentHistory);
+  const setDocumentHistory = useDocumentContext(
+    'RelationModalBody',
+    (state) => state.setDocumentHistory
+  );
 
   const [isConfirmationOpen, setIsConfirmationOpen] = React.useState(false);
-  const [isNavigating, setIsNavigating] = React.useState(false);
+  const [actionPosition, setActionPosition] = React.useState<'cancel' | 'back' | 'navigate'>(
+    'cancel'
+  );
 
   const [isModalOpen, setIsModalOpen] = React.useState(false);
+  // NOTE: Not sure about this relation modal context, maybe we should move this to DocumentContext?
```

Comment:
We can discuss if we want to move it or not (https://github.com/strapi/strapi/pull/23074#discussion_r1982985551)